Quark_Sketch/src/arduino/serial_print.c

```diff
@@ -0,0 +1,123 @@
+#include "serial_print.h"
+#include "arduino.h"
+
+
+#include <zephyr.h>
+#include "curie_shared_mem.h"
+
+#if defined(CONFIG_STDOUT_CONSOLE)
+#include <stdio.h>
+#define PRINT           printf
+#else
+#include <misc/printk.h>
+#define PRINT           printk
+#endif
+
+#define Rx_BUFF curie_shared_data->cdc_acm_shared_rx_buffer.data
+#define Rx_HEAD curie_shared_data->cdc_acm_shared_rx_buffer.head
+#define Rx_TAIL curie_shared_data->cdc_acm_shared_rx_buffer.tail
+#define Tx_BUFF curie_shared_data->cdc_acm_shared_tx_buffer.data
+#define Tx_HEAD curie_shared_data->cdc_acm_shared_tx_buffer.head
+#define Tx_TAIL curie_shared_data->cdc_acm_shared_tx_buffer.tail
+#define SBS     SERIAL_BUFFER_SIZE
+
+struct device *dev;
+
+void serial_print(const char *buf)
+{
+	int size = strlen(buf);
+	for(int i = 0; i < size; i++)
+	{
+		serial_write(buf[i]);
+		
+	}
+}
+
+void serial_println(const char *buf)
+{
+	int size = strlen(buf);
+	for(int i = 0; i < size; i++)
+	{
+		serial_write(buf[i]);
+		
+	}
+	serial_write('\r');
+	serial_write('\n');
+}
+
+void serial_printInt(int val)
+{
+	char buf[32];
+	memset(buf, '\0', 32);
+	int digits = 0;
+	int n = val;
+	while(n > 0)
+	{
+		digits++;
+		n /= 10;
+	}
+	
+	sprintf(buf, "%d", val);
+	serial_print(buf);
+}
+
+int serial_write(const uint8_t uc_data)
```

Comment:
It would be nice to provide a `serial_writec` for single char and `serial_writes` for a string. I know they can implement `serial_writes` themselves, but then they will use your `serial_write` for each character and the initial check on line 66 will happen for every character. We could eliminate that by implementing a `writes` function that just checks once before sending all the chars.